server/src/main/java/org/elasticsearch/transport/TransportService.java

```diff
@@ -275,6 +275,11 @@ public TransportService(
             clusterSettings,
             connectionManager,
             taskManger,
+            new ClusterSettingsLinkedProjectConfigService(
+                settings,
+                clusterSettings != null ? clusterSettings : ClusterSettings.createBuiltInClusterSettings(),
+                DefaultProjectResolver.INSTANCE
+            ),
```

Comment:
Maybe this is where we can use the `LinkedProjectConfigService.NOOP` when `clusterSettings == null`? I feel its behaviour is more aligned with the existing code, i.e. no settings consumer registrion and no update for remoteCluserService.